public/app/core/components/Select/FolderPicker.tsx

```diff
@@ -130,13 +136,17 @@ export class FolderPicker extends PureComponent<Props, State> {
   };
 
   createNewFolder = async (folderName: string) => {
-    // @ts-ignore
     const newFolder = await createFolder({ title: folderName });
     let folder = { value: -1, label: 'Not created' };
     if (newFolder.id > -1) {
       appEvents.emit(AppEvents.alertSuccess, ['Folder Created', 'OK']);
       folder = { value: newFolder.id, label: newFolder.title };
-      await this.onFolderChange(folder);
+      this.setState(
+        {
+          folder: newFolder,
+        },
+        () => this.props.onChange({ id: newFolder.value!, title: newFolder.label! })
+      );
```

Comment:
had to duplicate this so I wouldn't call `onFolderChange`. Could potentially be extracted to its own function to reduce the duplication.